internal/jobcontainers/logon.go

```diff
@@ -110,7 +110,10 @@ func (c *JobContainer) processToken(ctx context.Context, userOrGroup string) (wi
 	}
 
 	if groupExists(userOrGroup) {
-		userName = c.id[:winapi.UserNameCharLimit]
+		userName := c.id
+		if len(userName) > winapi.UserNameCharLimit {
+			userName = userName[:winapi.UserNameCharLimit]
```

Comment:
will this cause weird utf issues if the slice happens in the middle of a utf-8 rune, causing an invalid codepoint to be passed to windows?

Reply:
Yea, slicing a 4 byte emoji at [:3] would yield ���. Now I'm kind of curious what Windows classifies as a 20 character limit (e.g. 20 bytes or 20 utf16 entries regardless of width).

Reply:
But if we are planning to pass this to a Windows API we will anyway have to convert this from UTF-8 to UTF-16, right?